### PR TITLE
[BEAM-5303] Adding Python VR tests for Java Reference Runner.

### PR DESCRIPTION
**Pull request by youngoli**
Adding a gradle target and a test class for running ValidatesRunner
tests with the Java Reference Runner. Also had to add a dependency to
base_image_requirements.txt to get it to run properly. Note that this
was also causing an error when running Flink's VR tests with Docker.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by youngoli** [comment]
R: @robertwb 
CC: @HuangLED 

Robert, in addition to a regular review, can you confirm whether it's ok to add tenacity to the Docker container? It fails when trying to run tests from fn_api_runner_test otherwise, but I'm not sure if it's appropriate to add it to all built Docker containers if it's only needed for a single test.

**Comment by youngoli** [comment]
Run Python PreCommit

**Comment by youngoli** [comment]
Run Python Precommit

**Comment by robertwb** [comment]
Yes, it's fine to add this to the docker container.

File: sdks/python/apache_beam/runners/portability/java_reference_runner_test.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      ]
+
+    @classmethod
+    def get_runner(cls):
+      return portable_runner.PortableRunner()
+
+    def create_options(self):
+      options = super(JavaReferenceRunnerTest, self).create_options()
+      options.view_as(DebugOptions).experiments = ['beam_fn_api']
+      options._all_options['parallelism'] = 1
```
Comment (robertwb) [inline sdks/python/apache_beam/runners/portability/java_reference_runner_test.py]: Is this needed?

Comment (youngoli) [reply sdks/python/apache_beam/runners/portability/java_reference_runner_test.py]: Leaving this off seems to cause a test_sdf to fail. I haven't looked into the specifics of why yet (I'll do that when I get to implementing SDF), but the option seems necessary for now.

**Comment by youngoli** [comment]
@swegner for merging.

**Pull request merged**